### ISSUE TITLE
"paths" in jsconfig does not affect intellisense

### ISSUE THREAD

**Issue by jeffijoe** [Microsoft/vscode#32185]
![image](https://user-images.githubusercontent.com/2505474/29119069-f03d3f28-7d04-11e7-8677-0742477749e9.png)

- VSCode Version: Code 1.14.2 (cb82febafda0c8c199b9201ad274e25d9a76874e, 2017-07-19T23:22:50.650Z)
- OS Version: Darwin x64 16.7.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
EditorConfig|Edi|0.9.4
vscode-docker|Pet|0.0.16
vscode-eslint|dba|1.2.11
tslint|eg2|0.17.0
prettier-vscode|esb|0.22.1
Go|luk|0.6.63
csharp|ms-|1.11.0
sublime-keybindings|ms-|2.9.1
vscode-icons|rob|7.12.0
sass-indented|rob|1.4.1
vscode-coverage-gutters|rya|1.1.1

(2 theme extensions excluded)

---

Steps to Reproduce:

1. Set up a folder structure as outlined in the screenshot.
2. Use the following jsconfig:
```json
{
  "compilerOptions": {
    "experimentalDecorators": true,
    "baseUrl": "./",
    "module": "es6",
    "paths": {
      "features/*": ["src/client/features/*"],
      "components/*": ["src/client/components/*"],
    }
  }
}
```

3. Try to get autocompletion for the path aliases.

Reproduces without extensions: Yes

**Comment by mjbvz** [comment]
Looks like the suggest part is related to https://github.com/Microsoft/TypeScript/issues/14282 . If you manually type out the path, and then run `Go to definition` on the path, does it open the correct file?

Also, just want to make sure your actual `jsconfig.json` does not have the stray `,` on the line `["src/client/components/*"],`

**Comment by jeffijoe** [comment]
Removed the comma, no difference.

Go to definition on finished path:

![image](https://user-images.githubusercontent.com/2505474/29155357-f0a6ed1e-7d9b-11e7-883b-9802ab57686f.png)

**Comment by Rauttis** [comment]
I'm seeing the same issue, no file/folder suggestions for paths set up in jsconfig. However for me `Go to definition` is working correctly like @mjbvz suggests, it's just the suggestion part that is broken.

**Comment by Gywem** [comment]
The same issue here. It doesn't autocomplete neither the path nor the modules exported inside the file.

**Comment by mjbvz** [comment]
@Rauttis The suggestions issue for `paths` is being tracked by https://github.com/Microsoft/TypeScript/issues/19113

--- 

@jeffijoe Can you try with VSCode 1.17. If it still isn't working, two more things to try:

1. In the js file, run `JavaScript: Go to project configuration` command. This should open the `jsconfig.json` file. If it does not, you need to fix the `excludes` or `includes` in the `jsconfig.json`

2.  Try enabling [`// @ts-check`](https://code.visualstudio.com/Docs/languages/javascript#_type-checking-and-quick-fixes-for-javascript-files) in the file. This can help track down issues with module import paths

**Comment by jeffijoe** [comment]
@mjbvz I am on the latest and greatest stable VS Code 1.17 😄 

1. This opens the correct `jsconfig.json`.
2. Adding `// @ts-check` did not help either.

**Comment by mjbvz** [comment]
Try using:

```
      "module": "commonjs",
```

in your `jsconfig.json`. This is required to resolve import paths like `../thing-a` to `.../thing-a/index.js`

**Comment by jeffijoe** [comment]
@mjbvz that seems to get relative path intellisense working, but not the path aliases.

**Comment by mjbvz** [comment]
Sounds like https://github.com/Microsoft/TypeScript/issues/14282 

The intellisense from the imported modules still should work, the path suggestions for paths are just incorrect. Is that what you see?

**Comment by jeffijoe** [comment]
When I invoke Intellisense while inside the path string in the import statement, it takes like 10 secs of loading, then I only see node_modules paths. Sometimes it shows the path keys from the jsconfig, for example “components/*”. Does that help?

**Comment by mjbvz** [comment]
Yes the intellisense bug is Microsoft/TypeScript#14282 / Microsoft/TypeScript#19113

The performance issue was https://github.com/Microsoft/TypeScript/issues/16329 . It should be fixed in TypeScript 2.6 which is included in the current VS Code insiders builds. Please try this out and let me know if you still see the performance problems

**Comment by xahon** [comment]
@mjbvz why typescript if he uses js?

I have the same problem too on 1.18.0-insider
Intellisense works like a charm with "path-intellisense.mappings" but goto/peek definition not working
